src/app/core/views/Preferences/tabs/Account/AccountDetails.tsx

```diff
@@ -161,37 +176,165 @@ function AccountDetailsModal({
+
+  const [status, setStatus] = useState<
+    | { tag: 'ready' }
+    | { tag: 'loading' }
+    | { tag: 'error'; type: 'NAME_INVALID' | 'LASTNAME_INVALID' | 'PASSWORD_INVALID' | 'SAME_EMAIL' | 'UNKNOWN' }
+  >({ tag: 'ready' });
+
+  useEffect(() => {
+    if (isOpen) {
+      setStatus({ tag: 'ready' });
+    }
+  }, [isOpen]);
+
+  async function onChange(e) {
+    e.preventDefault();
+    if (email === newEmail) {
+      setStatus({ tag: 'error', type: 'SAME_EMAIL' });
+    } else {
+      try {
+        setStatus({ tag: 'loading' });
+        const correctPassword = await areCredentialsCorrect(email, password);
+        if (correctPassword) {
+          // TODO -> Send verificaion email
+          // Send verification to newEmail
+          notificationsService.show({
+            text: translate('views.account.tabs.account.accountDetails.changeEmail.sucessSendingVerification', {
+              email: newEmail,
+            }),
+            type: ToastType.Success,
+          });
+          onClose();
+        } else {
+          setStatus({ tag: 'error', type: 'PASSWORD_INVALID' });
+        }
+      } catch {
+        setStatus({ tag: 'error', type: 'UNKNOWN' });
+        notificationsService.show({
+          text: translate('views.account.tabs.account.accountDetails.changeEmail.errorSendingVerification'),
+          type: ToastType.Error,
+        });
+      }
+    }
+  }
+
+  return (
+    <Modal isOpen={isOpen} onClose={onClose}>
+      <form className="flex flex-col space-y-5" onSubmit={onChange}>
+        <h1 className="text-2xl font-medium text-gray-80">
+          {translate('views.account.tabs.account.accountDetails.changeEmail.title')}
+        </h1>
+
+        <div className="flex flex-col space-y-3">
+          <Input
+            disabled
+            label={translate('views.account.tabs.account.accountDetails.changeEmail.email') as string}
+            value={email}
+            name="email"
+          />
+          <Input
+            required
+            variant="email"
+            autoComplete="off"
+            label={translate('views.account.tabs.account.accountDetails.changeEmail.newEmail') as string}
+            onChange={setNewEmail}
+            accent={status.tag === 'error' && status.type === 'SAME_EMAIL' ? 'error' : undefined}
+            message={
+              status.tag === 'error' && status.type === 'SAME_EMAIL'
+                ? (translate('views.account.tabs.account.accountDetails.changeEmail.errorSameEmail') as string)
+                : undefined
+            }
+            name="newemail"
+          />
+          <Input
+            required
+            disabled={status.tag === 'loading'}
+            label={translate('views.account.tabs.account.accountDetails.changeEmail.password') as string}
+            variant="password"
+            onChange={setPassword}
+            accent={status.tag === 'error' && status.type === 'PASSWORD_INVALID' ? 'error' : undefined}
```

Comment:
I have seen several conditions that are repeated, you can extract them in const so that modifying or reading them is easier. What do you think?

`const isInvalidPasswordError = status.tag === 'error' && status.type === 'PASSWORD_INVALID';
`